File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let hash = {}

        for (let i = 0; i<nums.length; i++) {
            if (!hash[nums[i]]) hash[nums[i]]=0
            hash[nums[i]]++

            if (hash[nums[i]] > 1) {
                return true
            }
        }

        return false
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}

        for i in range(len(nums)):
            num = nums[i]
            potential = target - num

            if num in hash:
                return [hash[num], i] 
            hash[potential] = i

        print(hash)
        